src/drivers/cms-client.ts

```diff
@@ -224,6 +224,7 @@ export interface CodeModelFileGroup {
   includePath?: {path: string; isSystem?: boolean;}[];
   defines?: string[];
   sources: string[];
+  isGenerated: boolean;
```

Comment:
Why do we need to keep these interfaces in `cms-client.ts`?  Shouldn't they be replaced with the new ones in `codemodel-driver-interface.ts`?

Reply:
The code model interface contains only types that are needed by the extension. The interface types are very similar to the CMake ServerAPI types but are not identical. In the case of `isGenterated`, it was necessary to extend the cms client types to get the types easy converted into the code model types. 

I understand the driver classes as a mapper so that the ServerAPI types have a different meaning than the code model types. The FileAPI types have a different schema then the ServerAPI types. The code model types make it consistent for the CMake tools.